sssd_test_framework/utils/sssd.py

```diff
@@ -783,3 +784,42 @@ def autofs(self) -> None:
         """
         self.sssd.authselect.select("sssd")
         self.sssd.enable_responder("autofs")
+
+    def nslcd(
+        self,
+        provider: str | list[str] = "id",
+        role: Optional[MultihostRole] = None,
+        domain: str | None = None,
+    ):
+        """
+        :param provider: SSSD providers (id, auth, chpass, ...)
+        :type provider: str | list[str]
+        :param role: MultihostRole (LDAP server)
+        :type role: MultihostRole
+        :param domain: Existing domain name, defaults to None (= default domain)
+        :type domain: str | None, optional
+        """
+        if domain is None:
+            domain = self.sssd.default_domain
+
+        self.sssd.fs.write(
+            "/etc/pam.d/sssdproxyldap",
+            """
+                auth    required pam_ldap.so
+                account required pam_ldap.so
+                password required pam_ldap.so
+                session required pam_ldap.so
+                """,
+        )
+
+        self.sssd.fs.write(
+            "/etc/nslcd.conf", f"uid nslcd\ngid ldap\nuri " f"ldap://{role.host.hostname}\n", dedent=False
```

Comment:
For nsdcl ldap coffig it will work .

Reply:
Can you please test it or is the rest of the configuration specified somewhere? I don't understand how it will get the userPassword from the directory with no encryption and privileged access to read the attribute to be able to log in.

Reply:
[root@client sssd]# ssh user-1@localhost
The authenticity of host 'localhost (<no hostip for proxy command>)' can't be established.
ECDSA key fingerprint is SHA256:QC8XilGyktQmXT84um5KuQjwvYjMAMRWpxqwbhpNNHM.
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added 'localhost' (ECDSA) to the list of known hosts.
(user-1@localhost) Password: 
Last login: Fri Aug 11 10:30:28 2023 from ::1
-sh-5.2$

Reply:
[domain/test]
proxy_lib_name = ldap
debug_level = 0xFFF0
proxy_pam_target = sssdproxyldap
id_provider = proxy

Reply:
-sh-5.2$ cat /etc/nslcd.conf
uid nslcd
gid ldap
uri ldap://master.ldap.test
-sh-5.2$

Reply:
[root@client sssd]# cat sssd.conf 
[sssd]
config_file_version = 2
services = nss, pam
domains = test
debug_level = 0xfff0

[domain/test]
proxy_lib_name = ldap
debug_level = 0xFFF0
proxy_pam_target = sssdproxyldap
id_provider = proxy
auth_provider = proxy
proxy_fast_alias = true

[autofs]
debug_level = 0xfff0

[ifp]
debug_level = 0xfff0

[kcm]
debug_level = 0xfff0

[nss]
debug_level = 0xfff0

[pac]
debug_level = 0xfff0

[pam]
debug_level = 0xfff0

[ssh]
debug_level = 0xfff0

[sudo]
debug_level = 0xfff0[root@client sssd]# ssh user-1@localhost
(user-1@localhost) Password: 
Last login: Sat Aug 12 07:06:04 2023 from ::1
-sh-5.2$ cat /etc/nslcd.conf
uid nslcd
gid ldap
uri ldap://master.ldap.test
-sh-5.2$

Reply:
Can you tell me why it works without a bind user/password and/or ssl/tls please? I don't know, I want to understand why and how nslcd works works without them.

Reply:
@pbrezina can answer this he asked me to remove the base_dn also from config . According to him nslcd is able to find those stuff on its own